frontends/web/src/routes/account/account.tsx

```diff
@@ -49,324 +49,254 @@ const WithCoinTypeInfo = [
 
-  public componentDidMount() {
-    this.checkSDCards();
-    if (!this.props.code) {
+  const onAccountChanged = useCallback(() => {
+    if (!code || status === undefined || status.fatalError) {
       return;
     }
-    this.subscribe();
-    this.onStatusChanged();
-  }
-
-  public componentWillUnmount() {
-    unsubscribe(this.subscriptions);
-  }
-
-  public UNSAFE_componentWillReceiveProps(nextProps: Props) {
-    if (nextProps.code && nextProps.code !== this.props.code) {
-      this.setState({
-        status: undefined,
-        balance: undefined,
-        syncedAddressesCount: 0,
-        transactions: undefined,
-      });
-    }
-  }
-
-  public componentDidUpdate(prevProps: Props) {
-    if (!this.props.code) {
-      return;
-    }
-    if (this.props.code !== prevProps.code) {
-      this.onStatusChanged();
-      this.checkSDCards();
-      unsubscribe(this.subscriptions);
-      this.subscribe();
-    }
-    if (this.deviceIDs(this.props.devices).length !== this.deviceIDs(prevProps.devices).length) {
-      this.checkSDCards();
+    if (status.synced && status.offlineError === null) {
+      const currentCode = code;
+      Promise.all([
+        accountApi.getBalance(currentCode).then(newBalance => {
+          if (currentCode !== code) {
+            // Results came in after the account was switched. Ignore.
+            return;
+          }
+          setBalance(newBalance);
+        }),
+        accountApi.getTransactionList(code).then(newTransactions => {
+          if (currentCode !== code) {
+            // Results came in after the account was switched. Ignore.
+            return;
+          }
+          setTransactions(newTransactions);
+        })
+      ])
+        .catch(console.error);
+    } else {
+      setBalance(undefined);
+      setTransactions(undefined);
     }
-  }
-
-  private subscribe() {
-    this.subscriptions.push(
-      syncAddressesCount(this.props.code, (code, syncedAddressesCount) => {
-        if (code === this.props.code) {
-          this.setState({ syncedAddressesCount });
-        }
-      }),
-      statusChanged(this.props.code, () => this.onStatusChanged()),
-      syncdone(this.props.code, () => this.onAccountChanged()),
-    );
-  }
-
-  private checkSDCards() {
-    Promise.all(this.deviceIDs(this.props.devices).map(deviceID => {
-      switch (this.props.devices[deviceID]) {
-      case 'bitbox':
-        return getDeviceInfo(deviceID)
-          .then(({ sdcard }) => sdcard);
-      case 'bitbox02':
-        return apiGet(`devices/bitbox02/${deviceID}/check-sdcard`)
-          .then(sdcard => sdcard);
-      default:
-        return [];
-      }
-    }))
-      .then(sdcards => sdcards.some(sdcard => sdcard))
-      .then(hasCard => this.setState({ hasCard }))
-      .catch(console.error);
-  }
+  }, [code, status]);
 
-  private onStatusChanged() {
-    const code = this.props.code;
-    if (!code) {
+  const onStatusChanged = useCallback(() => {
```

Comment:
this should also be called in a useEffect depending on `code` -> in the previous version `onStatusChanged()` was called in `componentDidUpdate`